packages/widgets/src/renderwidget.ts

```diff
@@ -0,0 +1,225 @@
+/*-----------------------------------------------------------------------------
+| Copyright (c) 2014-2017, PhosphorJS Contributors
+|
+| Distributed under the terms of the BSD 3-Clause License.
+|
+| The full license is in the file LICENSE, distributed with this software.
+|----------------------------------------------------------------------------*/
+import {
+  Message, MessageLoop
+} from '@phosphor/messaging';
+
+import {
+  ISignal
+} from '@phosphor/signaling';
+
+import {
+  VirtualContent, VirtualDOM
+} from '@phosphor/virtualdom';
+
+import {
+  Widget
+} from './widget';
+
+
+/**
+ * An object which can be used as a model for a render widget.
+ */
+export
+interface IRenderModel {
```

Comment:
The name is not the same in both places, but your model is likely to wrap the store anyway.

We were talking about a few alternatives:
1) Don't have a model at all on this class, and let the subclass define it.
2) Make the model a simple `T | null` and have a `protected modelSwapped(old, new)` method which gets called, and let the subclass subscribe/unsubscribe as/if needed to their model type.